### ISSUE TITLE
I can't using "required: true" attribute with my $ref structure

### ISSUE THREAD

**Issue by jonnatasilva** [joelittlejohn/jsonschema2pojo#824]
I have tried use "required: true" with an object, this way:
```
{
  "type": "object",
  "properties": {
    "address": {
		"$ref": "address.json",
		"required": true
     }
  }
}
```
But I don't got a @NotNull in my object. I know If I use required["address"], I will have success, but I want use this way above.

**Comment by hiddenalpha** [comment]
Same problem here using 0.4.37.

I debugged into the code to further cut down the problem (Because I was not sure in which lib I needed to search that behavior).

While properties get collected they run trough PropertyRule.java. There [at line 77](https://github.com/joelittlejohn/jsonschema2pojo/blob/jsonschema2pojo-0.4.37/jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/PropertyRule.java#L77), the _node_ gets passed to [resolveRefs()](https://github.com/joelittlejohn/jsonschema2pojo/blob/jsonschema2pojo-0.4.37/jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/rules/PropertyRule.java#L152). This as I understand is responsible to load the referenced model in case the node refers to such one and returns a new node based on the loaded model. In case its not a reference, it simply returns the passed-in node.

The passed-in node wears our "required=true" property. But if that method returns a new instance, this will not have that property any longer. This may could be the problematic part.

**Comment by hiddenalpha** [comment]
#157, #397, #263 and #239 seem strongly related. Duplicates?

**Comment by joelittlejohn** [comment]
Until recently, it was explicitly defined in the JSON schema spec that all other fields in a schema should be ignored when $ref is used. This is why siblings of $ref are ignored.

I think this has changed in the most recent spec. Could you find a reference to the part that dictates what the behaviour should be now?

**Comment by hiddenalpha** [comment]
Didn't further research in JSON schema spec. So I've no idea if there something has changed.

For our project, first we discussed to switch to v4 of the spec. But as I understand there's no v4 support yet. (Is that correct?)

But then we discussed to add _required_ array into our schema-v3 file as specified in schema-v4. This idea rose up because in [comment at 397](https://github.com/joelittlejohn/jsonschema2pojo/issues/397#issuecomment-136195886), You wrote something about that this part of v4 spec would be supported (Could that work?).

Didn't start tinkering yet. But may we're able to solve that problem for our case. I'm working on other stuff right now. But may I'll have some time later to try that approach.
